qualtran/bloqs/basic_gates/rotation.py

```diff
@@ -325,6 +461,83 @@ def __str__(self):
         return f'Rz({self.angle})'
 
 
+@bloq_example
+def _rz() -> Rz:
+    a = sympy.Symbol('a')
+    rz = Rz(a)
+    return rz
+
+
+_RZ_DOC = BloqDocSpec(bloq_cls=Rz, examples=[_rz], call_graph_example=None)
+
+
+@frozen
+class CRz(Bloq):
+    r"""A controlled Rz rotation.
+
+    Given `angle` $\theta$, the unitary matrix of this gate is:
+    $$
+    C[R_Z(\theta)] =
+    \begin{bmatrix}
+        1 & & &  \\
+        & 1 & &  \\
+        & & e^{-i \theta/2} &  \\
+        & & &  e^{i \theta/2}
+    \end{bmatrix}
+    $$
+
+    Args:
+        angle: The rotation angle in radians.
+        eps: The precision of the rotation. This parameter is for bookkeeping and does
+            not affect e.g. the tensor representation of this gate. When synthesizing
+            a rotation from a discrete gate set, you must fix a precision `eps`.
+
+    Registers:
+        ctrl: Whether the rotation is active.
+        q: The qubit on which we optionally perform the rotation.
+
+    References:
+        [Elementary gates for quantum computation](https://arxiv.org/abs/quant-ph/9503016).
+        Barenco et al. 1995. Special case of Lemma 5.4.
+
+        [Is Controlled(𝑅𝑧(𝜃)) more expensive than Controlled(𝑍𝑡) on the surface code?](https://quantumcomputing.stackexchange.com/a/40012).
+        Adam Zalcman. 2024.
+    """
+
+    angle: SymbolicFloat
+    eps: SymbolicFloat = 1e-11
+
+    @cached_property
+    def signature(self) -> 'Signature':
+        return Signature.build(ctrl=1, q=1)
+
+    def build_composite_bloq(
+        self, bb: 'BloqBuilder', ctrl: 'Soquet', q: 'Soquet'
+    ) -> Dict[str, 'SoquetT']:
+        from qualtran.bloqs.basic_gates import CNOT
+
+        t = self.angle / np.pi
+        q = bb.add(ZPowGate(t / 2), q=q)
+        ctrl, q = bb.add(CNOT(), ctrl=ctrl, target=q)
+        q = bb.add(ZPowGate(-t / 2), q=q)
```

Comment:
Please propagate `eps` to the `ZPowGate`s as `self.eps / 2`

Reply:
done